### PR TITLE
feat: write account and storage hashes in genesis

### PR DESCRIPTION
**Pull request by Rjected**
This is a starting point for inserting hashed accounts / storage when inserting the genesis block. In hive we discovered that building an empty payload on top of genesis without running stages would result in an empty root, so we need to populate the correct tables in order to calculate the root.

### CODE REVIEW

**Review by gakonst — APPROVED** [review]
fine w me - @rkrasiuk ptal

File: bin/reth/src/drop_stage.rs [hunk 1]
```diff
@@ -74,9 +74,11 @@ impl Command {
                     tx.clear::<tables::StorageChangeSet>()?;
                     tx.clear::<tables::Bytecodes>()?;
                     tx.put::<tables::SyncStage>(EXECUTION.0.to_string(), 0)?;
-                    insert_genesis_state::<Env<WriteMap>>(tx, self.chain.genesis())?;
                     Ok::<_, eyre::Error>(())
                 })??;
+
+                let tx = db.tx_mut()?;
+                insert_genesis_state::<Env<WriteMap>>(&tx, self.chain.genesis())?;
```
[the remark this answers is not in the document]
Comment (rkrasiuk) [reply bin/reth/src/drop_stage.rs:81]: that's not right, we have to commit this

File: crates/staged-sync/src/utils/init.rs [hunk 2]
```diff
@@ -62,7 +67,12 @@ pub fn init_genesis<DB: Database>(
     let tx = db.tx_mut()?;
     insert_genesis_state::<DB>(&tx, genesis)?;
 
+    // use transaction to insert genesis header
+    let transaction = Transaction::new_raw(&db, tx);
+    insert_genesis_hashes(transaction, genesis)?;
+
     // Insert header
+    let tx = db.tx_mut()?;
```
Comment (rkrasiuk) [inline crates/staged-sync/src/utils/init.rs:75]: this should fail to acquire the write transaction because it's already owned by `Transaction`

**Review by onbjerg — APPROVED** [review]

**Review by rkrasiuk — CHANGES_REQUESTED** [review]

File: bin/reth/src/drop_stage.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rjected) [reply bin/reth/src/drop_stage.rs:81]: tbh not sure why I moved it out of the closure, just added it back which should commit

**Review by gakonst — APPROVED** [review]

**Pull request merged**